x-pack/plugins/code/server/lsp/proxy.ts

```diff
@@ -213,6 +214,7 @@ export class LanguageServerProxy implements ILanguageServerHandler {
 
   public connect(): Promise<MessageConnection> {
     if (this.clientConnection) {
+      this.registerOnNotificationHandler(this.clientConnection);
```

Comment:
connect() may be invoked multiple time, thus the registerOnNotification may also be invoked multiple times.
So this is not a good place to register the handler.
try add one on line 182